cmd/gcp-controller-manager/node_csr_approver.go

```diff
@@ -695,6 +697,45 @@ func validateInstanceGroupHint(instanceGroupUrls []string, instanceGroupHint str
 	return resolved, nil
 }
 
+var errNotFoundListReferrers = errors.New("not found the entry in ListReferrers")
+
+func checkInstanceReferrersBackOff(ctx *controllerContext, instance *compute.Instance, clusterInstanceGroupUrls []string) (bool, error) {
+	if !ctx.csrApproverListReferrersConfig.enabled {
+		return false, nil
+	}
+	klog.Infof("Using compute.InstancesService.ListReferrers to verify cluster membership of instance %q", instance.Name)
+
+	var found bool
+	startTime := time.Now()
+	backoffPolicy := wait.Backoff{
+		Duration: ctx.csrApproverListReferrersConfig.initialInterval,
+		Factor:   1.5,
+		Jitter:   0.2,
+		Steps:    ctx.csrApproverListReferrersConfig.retryCount,
+	}
+	webhook.WithExponentialBackoff(context.TODO(), backoffPolicy, func() error {
+		var retryErr error
+		found, retryErr = checkInstanceReferrers(ctx, instance, clusterInstanceGroupUrls)
+		if retryErr != nil {
+			return retryErr
+		} else if !found {
+			return errNotFoundListReferrers
+		}
+		return nil
+	},
+		func(err error) bool {
+			return err != nil
+		},
+	)
+
+	if found {
+		klog.V(2).Infof("Determined cluster membership of instance %q using compute.InstancesService.ListReferrers after %v", instance.Name, time.Since(startTime))
+	} else {
+		klog.Warningf("Could not determine cluster membership of instance %q using compute.InstancesService.ListReferrers after %v; falling back to checking all instance groups", instance.Name, time.Since(startTime))
```

Comment:
should we also add a latency metric?

Reply:
Each of the calls is covered by a dedicated metric, should we add a e2e one as well?
There is an e2e latency metric for the whole approval process, I don't see an appropriate metric that could be used to reflect multiple-rpc-calls, should I create a new one?